web/packages/teleterm/src/ui/AppUpdater/AutoUpdatesManagement.tsx

```diff
@@ -0,0 +1,321 @@
+  // A local state allows us to unselect the checkbox without choosing any managing cluster.
+  const [localIsAutoManaged, setLocalIsAutoManaged] = useState(isAutoManaged);
+
+  const isMostCompatibleCheckboxDisabled =
+    isCheckingForUpdates ||
+    (autoUpdatesStatus.enabled === false &&
+      autoUpdatesStatus.reason === 'no-compatible-version');
+
+  const options = makeOptions(autoUpdatesStatus, getClusterName);
+
+  return (
+    <>
+      {options.length > 1 && (
+        <Stack width="100%">
+          <label
+            css={`
+              gap: ${p => p.theme.space[1]}px;
+              display: flex;
+            `}
+          >
+            <CheckboxInput
+              checked={localIsAutoManaged}
+              disabled={isMostCompatibleCheckboxDisabled}
+              onChange={e => {
+                setLocalIsAutoManaged(e.target.checked);
+                if (e.target.checked) {
+                  changeManagingCluster(undefined);
+                }
+              }}
+            />
+            Use the most compatible version from your clusters
+          </label>
+          {!localIsAutoManaged && (
+            <>
+              <Text>Select a cluster to manage updates:</Text>
+              <RadioGroup
+                gap={0}
+                name="managingCluster"
+                size="small"
+                value={
+                  autoUpdatesStatus.enabled &&
+                  autoUpdatesStatus.source.kind === 'managing-cluster' &&
+                  autoUpdatesStatus.source.clusterUri
+                }
+                onChange={clusterUri => {
+                  setLocalIsAutoManaged(false);
+                  changeManagingCluster(clusterUri);
+                }}
+                options={options}
+              />
+            </>
+          )}
+          <hr
+            css={`
+              border-color: ${props => props.theme.colors.spotBackground[2]};
+              border-style: solid;
+              width: 100%;
+              margin-bottom: 0;
+              margin-top: ${p => p.theme.space[2]}px;
+            `}
+          />
+        </Stack>
+      )}
+    </>
+  );
+}
+
+function makeOptions(
+  status: AutoUpdatesStatus,
+  getClusterName: (clusterUri: RootClusterUri) => string
+) {
+  let source:
+    | {
+        clusters: Cluster[];
+        unreachableClusters: UnreachableCluster[];
+      }
+    | undefined;
+
+  if (status.enabled === false) {
+    if (status.reason === 'no-compatible-version') {
+      source = status;
+    }
+  } else {
+    if (
+      status.source.kind === 'most-compatible' ||
+      status.source.kind === 'managing-cluster'
+    ) {
+      source = status.source;
+    }
+  }
+
+  if (!source) {
+    return [];
+  }
+
+  const candidateClusters = source.clusters
+    .filter(c => c.toolsAutoUpdate)
+    .map(c => {
+      const otherNames = c.otherCompatibleClusters.map(
+        c => `"${getClusterName(c)}"`
```

Comment:
I don't really like putting cluster name in quotations. :D I feel like we do this on the backend a lot because in Go it's so easy to just do `%q`. But in the UI it looks a bit off to me. What do you think about dropping the quotations and just showing plain cluster names.